ledger/metrics/src/main/scala/com/daml/metrics/api/opentelemetry/OpenTelemetryFactory.scala

```diff
@@ -57,62 +62,75 @@ trait OpenTelemetryFactory extends Factory {
 
   override def gaugeWithSupplier[T](
       name: MetricName,
-      gaugeSupplier: () => () => (T, MetricsContext),
+      valueSupplier: () => T,
+  )(implicit
+      context: MetricsContext = MetricsContext.Empty
   ): Unit = {
-    val valueSupplier = gaugeSupplier()
-    valueSupplier()._1 match {
+    val value = valueSupplier()
+    val attributes = context.asAttributes
+    value match {
       case _: Int =>
         otelMeter
           .gaugeBuilder(name)
           .ofLongs()
           .buildWithCallback { consumer =>
-            val (value, context) = valueSupplier()
-            consumer.record(value.asInstanceOf[Int].toLong, context.asAttributes)
+            val value = valueSupplier()
+            consumer.record(value.asInstanceOf[Int].toLong, attributes)
           }
         ()
       case _: Long =>
         otelMeter
           .gaugeBuilder(name)
           .ofLongs()
           .buildWithCallback { consumer =>
-            val (value, context) = valueSupplier()
-            consumer.record(value.asInstanceOf[Long], context.asAttributes)
+            val value = valueSupplier()
+            consumer.record(value.asInstanceOf[Long], attributes)
           }
         ()
       case _: Double =>
         otelMeter
           .gaugeBuilder(name)
           .buildWithCallback { consumer =>
-            val (value, context) = valueSupplier()
-            consumer.record(value.asInstanceOf[Double], context.asAttributes)
+            val value = valueSupplier()
+            consumer.record(value.asInstanceOf[Double], attributes)
           }
         ()
       case _ =>
       // NoOp as opentelemetry only supports longs and doubles
     }
   }
-  override def meter(name: MetricName): Meter = OpentelemetryMeter(
+  override def meter(name: MetricName)(implicit
+      context: MetricsContext = MetricsContext.Empty
+  ): Meter = OpentelemetryMeter(
     name,
     otelMeter.counterBuilder(name).build(),
+    context,
   )
-  override def counter(name: MetricName): MetricHandle.Counter = OpentelemetryCounter(
+  override def counter(name: MetricName)(implicit
+      context: MetricsContext = MetricsContext.Empty
+  ): MetricHandle.Counter = OpentelemetryCounter(
     name,
     otelMeter.upDownCounterBuilder(name).build(),
+    context,
   )
-  override def histogram(name: MetricName): MetricHandle.Histogram = OpentelemetryHistogram(
+  override def histogram(name: MetricName)(implicit
+      context: MetricsContext = MetricsContext.Empty
+  ): MetricHandle.Histogram = OpentelemetryHistogram(
     name,
     otelMeter.histogramBuilder(name).ofLongs().build(),
+    context,
   )
 }
 
-case class OpentelemetryTimer(name: String, histogram: LongHistogram) extends Timer {
+case class OpentelemetryTimer(name: String, histogram: LongHistogram, timerContext: MetricsContext)
```

Comment:
Consistency: OpenTelemetry is spelled camel-cased, hence, I'd strongly suggest renaming all occurrences to follow that:
```suggestion
case class OpenTelemetryTimer(name: String, histogram: LongHistogram, timerContext: MetricsContext)
```
You may want to do this in a separate PR if it has a large impact radius.

Reply:
Please see already existing PR with this change: https://github.com/digital-asset/daml/pull/15357